doc/vlan/switchport-mode-support/Switchport Mode and VLAN CLI Enhancement.md

```diff
@@ -0,0 +1,760 @@
+
+#### Configuring Port & PortChannel from Routed to Access 
+
+In these examples, Ethernet0 will be configured as “Access” from “Routed”. We will be using some of exisitng Vlans (Vlan2 to Vlan12 created in the above section) for vlan member configuration on Port. 
+
+By default, all ports have IP assigned. For Switchport configuration we have to remove IP assignment otherwise CLI will show following Error:
+
+**For Ethernet0**
+
+<img src="https://user-images.githubusercontent.com/61490193/236294097-1c79ee78-0b83-4daf-ae48-39de60804683.png" width="65%" height="30%">
+
+
+Following Steps will be taken to configure “Access”  mode  on Ethernet0 from “Routed”.
+
+**1.** Remove IP assigned on Ethernet0
+
+```
+admin@sonic:~$ sudo config interface ip remove Ethernet0 10.0.0.0/31
+```
+
+**2.** Configure Ethernet0 as Access
+
+```
+admin@sonic:~$ sudo config switchport mode access Ethernet0        
+```
+**3.** View current configuration by using “show interfaces switchport status” command
+
+```
+admin@sonic:~$ show int switchport status
+```
+<img src="https://user-images.githubusercontent.com/61490193/236170364-27cc8f1e-a6f3-401a-90e5-8ed035b0e65e.png" width="65%" height="30%">
+
+
+**4.** Untagged Vlan Member Assignment on Access Port
+
+```
+admin@sonic:~$ sudo config vlan member add 2 -u Ethernet0
+```
+
+<img src="https://user-images.githubusercontent.com/61490193/236294143-3df37252-dcf4-46d2-bb0d-b8ad104dbb66.png" width="65%" height="30%"> 
+
+**5.** Multiple Untagged Vlan Member Assignment on Access Port
+
+Ethernet0 is in Access mode, it can have only 1 untagged member. Configuring More than 1 untagged member on Access Port will show following error:
+
+<img src="https://user-images.githubusercontent.com/61490193/236170322-030f56ac-b829-4273-95c1-193ddd33807a.png" width="65%" height="30%"> 
+
+**6.** Tagged Vlan Member Assignment on Access Port
+
+Ethernet0 is in Access mode, it cannot have tagged members. Configuring tagged member on Access Port will show following error:
+
+<img src="https://user-images.githubusercontent.com/61490193/236170445-2d086077-1c78-45c6-9e23-f03b1b280a14.png" width="65%" height="30%"> 
+
+**7.** IP Assignment on Access Port
+
+Ethernet0 is in Access mode, IP assignment on the access port is not allowed. Configuring IP Assignment on Access Port will show following error:
+
+<img src="https://user-images.githubusercontent.com/61490193/236170194-2423f230-483a-42b7-936c-5939bf0e84ae.png" width="65%" height="30%"> 
+
+**8.** Change Mode from Access to Routed
+
+Ethernet0 is in Access mode, switching an access port to routed is not possible until it has an untagged member. Changing  mode from access to routed will show following error:
+
```

Comment:
When changing from aCcess to routed, what is in the confiogDB for the corresponding port?  will it remove the "mode" completely or have "mode: routed"?  Can you clarify on this?  I would prefer the "mode" be removed completely...

Reply:
when changing from access to routed, the mode of the corresponding port in config_db will be changed from "access" to "routed". It cannot be removed completely as this "mode" attribute is saving state of the port that is either routed , access or trunk. 
This was discussed in our internal discussion and it was decided that this attribute will not be removed completely for changing from access to routed or from trunk to routed.

Reply:
I see... Please spell this out in your doc as I have asked this before and your previous response is yes you will modify to remove the "mode" when transition from "access" or "trunk" to "routed". But now you are reversing what you have previously agreed to... So please specify this clearly in your doc so there is no ambiguity.

Reply:
So an interface in mode "routed" means it has no VLAN membership. it may not have ip address configured either and still qualify as mode "routed" correct?   If you agree this is correct statement, can you add this in the doc as well?

Reply:
I have updated this concern and explicitly spelled it out to avoid any ambiguity.

Reply:
Yes this is the correct statement, an interface in "routed" mode implies that it does not have VLAN membership. It's possible for an interface in "routed" mode not to have an IP address configured. This statement is added to provide further clarity on the behavior of interfaces in "routed" mode. Please see HLD example section **point 8** under "**Configuring Port & PortChannel from Routed to Trunk"** & **point 8** under **"Configuring Port & PortChannel from Routed to Access"** for specific example on this behavior.